lib/common/mem.h

```diff
@@ -80,6 +80,26 @@ extern "C" {
 #endif
 
 
+/*-**************************************************************
+*  Alignment check
+*****************************************************************/
+
+/* C11 support */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)
+
+#include <stdalign.h>
+# define MEM_ALIGN_COND(T) alignof(T)
+
+#elif defined(__GNUC__) || defined(_MSC_VER)
+
+# define MEM_ALIGN_COND(T) __alignof(T)
+
+#else
+
+# define MEM_ALIGN_COND(T) 1  /* most likely incorrect, just to pass tests */
```

Comment:
Won't this break some of the alignment in `huf_compress.c` on non-supported compilers?

Could we maybe default to `MIN(roundUpToPow2(sizeof(T)), sizeof(size_t))`?